qark/modules/findPending.py

```diff
@@ -21,456 +22,499 @@
-													found=True
-													if hasattr(v.initializer,'arguments'):
-														for a in v.initializer.arguments:
-															if type(a) is m.ClassLiteral:
-																explicit=True
-																results=[found,explicit]
-																return results
-													results=[found,False]
-													return results
-
-					elif type(c) is m.VariableDeclarator:
-						if hasattr(c.initializer,'type'):
-							if hasattr(c.initializer.type.name,'value'):
-								if c.initializer.type.name.value == "Intent":
-									if str(c.variable.name) == str(intent):
-										found=True
-										if hasattr(c.initializer,'arguments'):
-											for a in c.initializer.arguments:
-												if type(a) is m.ClassLiteral:
-													explicit=True
-													results=[found,explicit]
-													return results
-										explicit=False
-										results=[found,explicit]
-										return results
-						else:
-							results=intent_digger(c.initializer,intent)
-							if results[0] == True:
-								return results
-
-					else:
-						results=intent_digger(c,intent)
-						if results[0] == True:
-							return results
-
-			elif type(current) is m.VariableDeclaration:
-				if hasattr(current,'type'):
-					if type(current.type) is m.Type:
-						if hasattr(current.type.name,'value'):
-							if current.type.name.value == "Intent":
-								for v in current.variable_declarators:
-									if str(v.variable.name) == str(intent):
-										found=True
-										for a in v.initializer.arguments:
-											if type(a) is m.ClassLiteral:
-												explicit=True
-												results=[found,explicit]
-												return results
-			elif type(current) is m.VariableDeclarator:
-				if hasattr(c.initializer,'type'):
-					if hasattr(c.initializer.type.name,'value'):
-						if c.initializer.type.name.value == "Intent":
-							if str(c.variable.name) == str(intent):
-								found=True
-								for a in c.initializer.arguments:
-									if type(a) is m.ClassLiteral:
-										explicit=True
-										results=[found,explicit]
-										return results
-								explicit=False
-								results=[found,explicit]
-								return results
-			else:
-				if hasattr(current,'_fields'):
-					for r in current._fields:
-						results=intent_digger(getattr(current,r),intent)
-						if results[0] == True:
-							return results
-							
-	elif type(t) is list:
-		for y in t:
-			results=intent_digger(y,intent)
-			if results[0] == True:
-				return results
-
-	return results
+def start(queue, height):
+    """ Start finding pending intents """
+    results = []
+    global tree
+    global current_file
+    count = 0
+
+    # TODO - Look for use of fillIn method which can make this a much more exploitable condition
+    for j in common.java_files:
+        count = count + 1
+        pub.sendMessage('progress', bar='Pending Intents', percent=round(count * 100 / common.java_files.__len__()))
+        current_file = j
```

Comment:
this seems extraneous
I think you could just do `for current_file in common.java_files`, which will lead to less variables floating around.